OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -20,29 +20,71 @@ public class AttackMoveActivity : Activity
 	{
 		const int ScanInterval = 7;
 
+		readonly CPos destination;
+		readonly int nearEnough;
 		Activity inner;
-		int scanTicks;
+		Activity attack;
 		AutoTarget autoTarget;
+		bool moving;
 
-		public AttackMoveActivity(Actor self, Activity inner)
+		public AttackMoveActivity(Actor self, CPos destination, int nearEnough)
 		{
-			this.inner = inner;
+			this.destination = destination;
+			this.nearEnough = nearEnough;
 			autoTarget = self.TraitOrDefault<AutoTarget>();
+			moving = false;
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (autoTarget != null && --scanTicks <= 0)
+			if (IsCanceled)
 			{
-				autoTarget.ScanAndAttack(self, true);
-				scanTicks = ScanInterval;
+				if (inner != null)
+				{
+					inner = ActivityUtils.RunActivity(self, inner);
+					return this;
+				}
+
+				return NextActivity;
+			}
+
+			if (autoTarget == null)
+				return self.Trait<IMove>().MoveTo(destination, nearEnough);
+
+			if (attack == null)
+			{
+				var target = autoTarget.ScanForTarget(self, true);
+				if (target.Type != TargetType.Invalid)
+				{
+					if (inner != null)
+						inner.Cancel(self);
+
+					var attackBases = autoTarget.ActiveAttackBases;
+					foreach (var ab in attackBases)
+					{
+						if (attack == null)
+							attack = ab.GetAttackActivity(self, target, true, false);
```

Comment:
This will also need to call `OnQueueAttackActivity` once #16110 is merged.

Reply:
#16110 has been merged, so can you please add this now?

Reply:
Okay. added a call to `OnQueueAttackActivity` and rebased.